### DISCUSSION TITLE
Configure environment variables from secrets in applications

### DISCUSSION THREAD

**Discussion by nicola88** [mia-platform/community#489] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

12.4.0

### Description

Hi, we are trying to build a new application involving several microservices that use some providers credentials (AWS, Google, etc.) and we would like to configure the environment variables to pull these credentials from secrets configured with mlp, like the Console [allows to do manually](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/services#environment-variable-configuration), but we couldn't figure out from the documentation if this is feasible in the application JSON configuration.

**Comment by malta895** [comment, marked as answer]
Hi @nicola88,
what you are asking for is feasible, and the marketplace documentation is actually incomplete, thank you!

You can do it in any marketplace item, regardless of the type (Application, Example, Template, Plugin).

The `resources.services.defaultEnvironmentVariables` should be added an object with `"valueType": "secret"`; here is an example:
```json
{
          "name": "SOME_ENV",
          "valueType": "secret",
          "secretName": "someName",
          "secretKey": "secret.key"
}
```
The values have to be compiled as if they were inserted manually from the Console, following the indications of the page you linked.

In the meantime we will take care of updating the Marketplace documentation.
Hope this helps!

**Reply by malta895** [reply]
Since secrets are to be created on Kubernetes to be used, we suggest that the documentation of your Application clearly specifies that the user needs to take care of the secret creation, with the correct name and key.

**Reply by nicola88** [reply]
Thanks for the information, we will start updating our application as soon as possible.
